src/based-rematch.cpp

```diff
@@ -0,0 +1,170 @@
+#include <iostream>
+#include <algorithm>
+#include <cmath>
+
+#include "./include/puzzle.h"
+#include "./include/utils.h"
+
+/*** define puzzle sizes ***/
+#define NUM_VALUES 11
+#define VAL_SZ 4
+
+
+/*** define indexes into values vector based on the base ***/
+#define S8_START 0
+#define S8_END 7
+
+#define G16_START 8
+#define G16_END 23
+
+#define M36_START 24
+#define M36_END 59
+
+
+
+/*  
+*   indices 
+*  0-7 -> S8
+*  8-23 -> G16
+*  24-59 -> M38
+*/
+static const std::vector<std::string> base_nums = {"!", "@", "#", "$", "%", "^", "&", "*", "{", "}", "[", "]", \
+    "(", ")", "`", ">", "a", "b", "c", "d", "e", "f", "g", "h", "|", "~", ",", "/", "?", "_", "+", "=", "-", \
+    ":", "A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z"};
+
+
+//set indices into base_nums vector
+void set_start_end(int base, int *start, int *end)
+{
+  switch(base)
+  {
+    case 8:
+      *start = S8_START;
+      *end = S8_END;
+      break;
+    case 16:
+      *start = G16_START;
+      *end = G16_END;
+      break;
+    case 36:
+      *start = M36_START;
+      *end = M36_END;
+      break;
+  }
+}
+
+//return the index of the symbol in relation to its base from base_nums
+int find_val(int base, std::string symbol)
+{
+  int start, end;
+  set_start_end(base, &start, &end);
+  int i = 0;
+  for(; start <= end; start++) {
```

Comment:
make for loop: `for (int i = 0; start <= end; start++)`

Reply:
done